dev/com.ibm.ws.persistence/src/com/ibm/wsspi/persistence/internal/DatabaseStoreImpl.java

```diff
@@ -738,6 +781,52 @@ else if (suspendedLTC != null)
         return dbProductName;
     }
 
+    /**
+     * Attempt to ascertain if the given isolation level is supported on the given datasource
+     *<p>
+     *{@link java.sql.DatabaseMetaData#supportsTransactionIsolationLevel()}
+     *
+     * @param dataSource data source instance.
+     * @param isolationLevel valid isolation level value.
+     * @return 'true' if the driver DatabaseMetaData indicates the given transaction isolation is supported; 'false' otherwise
+     */
+    private boolean supportsIsolationLevel(WSDataSource dataSource, int isolationLevel) {
+        try {
+            // Query the metadata under a new transaction and commit right away
+            LocalTransactionCurrent localTranCurrent = this.localTranCurrent;
+            LocalTransactionCoordinator suspendedLTC = localTranCurrent.suspend();
+            EmbeddableWebSphereTransactionManager tranMgr = this.tranMgr;
+            Transaction suspendedTran = suspendedLTC == null ? tranMgr.suspend() : null;
+            boolean tranStarted = false;
+            try {
+                tranMgr.begin();
+                tranStarted = true;
+                Connection con = dataSource.getConnection();
+                try {
+                    return con.getMetaData().supportsTransactionIsolationLevel(isolationLevel);
+                } finally {
+                    con.close();
+                }
+            } finally {
+                try {
+                    if (tranStarted)
+                        tranMgr.commit();
```

Comment:
Are we changing anything here?  If not, might be better to rollback than commit.

Reply:
Sound fine to me